tools/python/toolchain.bzl

```diff
@@ -14,6 +14,7 @@
 
 """Definitions related to the Python toolchain."""
 
+load("@local_config_platform//:constraints.bzl", "HOST_CONSTRAINTS")
```

Comment:
Whoa, what is this magic? What else is in this local_config_platform? This seems useful, though I'm not sure how yet...

Reply:
Just this, but it is very useful for e.g. marking rules that should always resolve toolchains for the host via `exec_compatible_with`. We use this trick in rules_go to make `bazel run @rules_go//go` always run the correct `go` tool even with `--platforms`.

Reply:
The magic is in [`LocalConfigPlatformFunction`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java), but it's really easy (and could be fairly trivially migrated to a Starlark repo rule in the builtins, I bet).

The repo creates the following files:
- `WORKSPACE` and `MODULE.bazel`: Mark the repository (and depend on platforms 0.7.0).
- `BUILD.bazel`: Declares the `host` platform, using the `HOST_CONSTRAINTS` from `constraints.bzl`
- `constraints.bzl`: Creates the `HOST_CONSTRAINTS`, based on whatever Bazel autodetects as the host environment.